irispy/iris_tools.py

```diff
@@ -212,6 +220,102 @@ def get_iris_response(pre_launch=False, response_file=None, response_version=Non
+    size_cal_coeffs = cal_coeffs.shape
+
+    if size_time_cal_coeffs[1] != 2 or size_cal_coeffs[1] < 2:
+        # Raise ValueError as time coefficient have the wrong format.
+        raise ValueError("Incorrect number of elements either in time_cal_coeffs or in cal_coeffs.")
+
+    # Some time transformations.
+    # Convert the time_cal_coeffs given in the .geny file into a ``datetime.datetime`` object
+    # called t_cal_coeffs, so that the time differences will be in days...
+    t_cal_coeffs_flat = time_cal_coeffs.flatten()
+    t_cal_coeffs = [parse_time(x, format = 'utime') for x in t_cal_coeffs_flat]
+    t_cal_coeffs = np.array(t_cal_coeffs).reshape(size_time_cal_coeffs)
+
+    # For loop for carrying out the least-squares fit and computation of fit output.
+    count = 0
+    for i in time_obs:
+        aux_cal_coeffs = np.zeros(2*size_time_cal_coeffs[0])
+
+        # Exponent for transition between exp.decay intervals.
+        transition_exp = 1.5
+
+        fit_out = np.zeros(len(list(time_obs)), dtype=np.float64)
+
+        # Looking for the closest time in the calibration time intervals.
+        # Differences are given in years before passing to the next stage.
+        t_diff = time_obs - t_cal_coeffs
+
+        t_diff = t_diff.flatten()
+        
+        t_diff = [x.to(u.year) for x in t_diff]
+        
+        # To Convert to an array, quantities need to be dimensionless, hence dividing out the unit.
+        t_diff = np.array([x/u.year for x in t_diff])
+
+        w = np.where(t_diff < 0)[0][0]
+
+        # If the t_obs is betweeen the calibration time intervals of a
+        # calibration file (w % !=0) then the aux_coeffs are given by an
+        # exponential (coefficient and exponential value).
+        # If the t_obs is between the end calibration time interval of
+        # a calibration file (cal_file_t) and the beginning calibration
+        # time interval of the next calibration file (cal_file_t+1)
+        # (w % 2 == 0) then, the aux_coeffs are given by 4 values
+        # corresponding to a partial exponential obtained from
+        # cal_file_t and a complemetary exponential obtained from the
+        # cal_file_t+1
+        if w % 2 != 0:  # I.e., if w is not even...
+            dtt_0 = 1.
+            exp_0 = np.exp(cal_coeffs[w//2, 2]*(t_diff[w-1]))
+            aux_cal_coeffs[w-1:w+1] = np.array([dtt_0, dtt_0*exp_0])
+        else:
+            dtt_1 = (t_diff[w-1] / (t_diff[w-1] - t_diff[w]))**transition_exp
+            dtt_0 = 1. - dtt_1
+            exp_0 = np.exp(cal_coeffs[(w//2)-1, 2]*(t_diff[w-2]))
+            exp_1 = np.exp(cal_coeffs[w//2, 2]*(t_diff[w]))
+            aux_cal_coeffs[w-2:w+2] = np.array([dtt_0, dtt_0*exp_0, dtt_1, dtt_1*exp_1])
+        # print(aux_cal_coeffs)
+        # print(cal_coeffs[:,:2].reshape((aux_cal_coeffs.shape[0])))
+        fit_out[count] = np.matmul(aux_cal_coeffs, cal_coeffs[:, :2].reshape((aux_cal_coeffs.shape[0])))
+        count += 1
```

Comment:
Consistent with required change raised in above comment, delete ```count +=1``` and replace ```fit_out[count]``` with ```fit_out[i]```